sdk/core/Azure.Core.Amqp/src/AmqpMessageBody.cs

```diff
@@ -121,5 +151,40 @@ public bool TryGetSequence(out IEnumerable<IList<object>>? sequence)
             sequence = null;
             return false;
         }
+
+        private static void ValidateAmqpPrimitive(object value, string paramName)
+        {
+            Argument.AssertNotNull(value, paramName);
+            switch (value)
+            {
+                case string:
+                case byte:
+                case sbyte:
+                case char:
+                case short:
+                case ushort:
+                case int:
+                case uint:
+                case long:
+                case ulong:
+                case float:
+                case double:
+                case decimal:
+                case bool:
+                case Guid:
+                case DateTime:
+                    return;
+                case IEnumerable enumerable:
```

Comment:
Since the enumerable is passed in, shouldn't assume that it is safe to enumerate more than once.   Given the potential nesting here, this is tricky;  I don't know that we can reasonably safeguard by unrolling into a list or other known-safe collection.   

Maybe the best we can reasonably do is to explicitly call out the contract when the body is created (explicitly or implicitly) to detail that we expect that any enumerable types are expected to be safe to enumerate multiple times?